local/item.go

```diff
@@ -70,6 +72,48 @@ func (i *item) Open() (io.ReadCloser, error) {
 	return os.Open(i.path)
 }
 
+// LimitedReadCloser wraps io.LimitedReader and exposes the Close() method.
+type LimitedReadCloser struct {
```

Comment:
Can we un-export this type? The method returning io.ReadCloser it's probably enough. We can keep the type private? - yes.